### PR TITLE
Feature poolliquidity autofinding

### PR DESCRIPTION
**Pull request by DeFiDev**
Added the auto-selection wallet accounts feature for addpoolliquidity RPC method.
User should set single entry in the 'from' param, where "*" (star) is a key, and string/array of token amounts is a value.

### CODE REVIEW

File: src/masternodes/mn_rpc.cpp [hunk 1]
```diff
@@ -3081,39 +3225,174 @@ UniValue isappliedcustomtx(const JSONRPCRequest& request) {
...
+    // encode
+    CDataStream markedMetadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    markedMetadata << static_cast<unsigned char>(CustomTxType::AnyAccountsToAccounts)
+                   << msg;
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(markedMetadata);
+
+    if (scriptMeta.size() > nMaxDatacarrierBytes) {
+        throw JSONRPCError(RPC_VERIFY_REJECTED, "The output custom script size has exceeded the maximum OP_RETURN script size."
+                                                "It could have happened because too many \"from\" or \"to\" accounts balances."
```
Comment (ShengguangXiao) [inline src/masternodes/mn_rpc.cpp:3309]: "It could have happened" change to "It may"

File: src/masternodes/mn_rpc.cpp [hunk 2]
```diff
@@ -3081,39 +3225,174 @@ UniValue isappliedcustomtx(const JSONRPCRequest& request) {
...
+                                                "If you used autofinding, you can try to use \"pie\" selection mode for decreasing accounts count.");
```
Comment (ShengguangXiao) [inline src/masternodes/mn_rpc.cpp:3310]: "used" change to "use"

Comment (DeFiDev) [reply src/masternodes/mn_rpc.cpp:3310]: Which option do you think is better?

File: src/masternodes/mn_rpc.cpp (hunk 1 above)
Comment (DeFiDev) [reply src/masternodes/mn_rpc.cpp]: Done

File: src/masternodes/mn_rpc.cpp (hunk 2 above)
Comment (DeFiDev) [reply src/masternodes/mn_rpc.cpp]: Done

**Comment by monstrobishi** [comment]
This includes both sendtokenstoaddress and add pool liquidity from any amount

**Pull request merged**